### PR TITLE
enhancement(kubernetes_logs source): Expose `oldest_first`

### PR DESCRIPTION
**Pull request by jszwedko**
In response to https://github.com/vectordotdev/vector/issues/18088#issuecomment-1690491190

Might close: https://github.com/vectordotdev/vector/issues/18088

Signed-off-by: Jesse Szwedko <jesse.szwedko@datadoghq.com>

### CODE REVIEW

**Comment by akutta** [comment]
May want to mention the difference in behaviour for `max_read_bytes` based on the new option.

File: src/sources/kubernetes_logs/mod.rs [hunk 1]
```diff
@@ -171,6 +171,10 @@ pub struct Config {
     #[configurable(metadata(docs::type_unit = "bytes"))]
     max_read_bytes: usize,
 
+    /// Instead of balancing read capacity fairly across all watched files, prioritize draining the oldest files before moving on to read data from younger files.
```
Comment (neuronull) [inline src/sources/kubernetes_logs/mod.rs:174]: **💭 thought:** ‏"younger" -> "newer" ?

**Review by neuronull — APPROVED** [review]

File: src/sources/kubernetes_logs/mod.rs (hunk 1 above)
Comment (jszwedko) [reply src/sources/kubernetes_logs/mod.rs:174]: I think "younger" here is also correct 🤔 I copy/pasted this description from the `file` source.

**Comment by jszwedko** [comment]
> May want to mention the difference in behaviour for `max_read_bytes` based on the new option.

Good thought, I modified the description for that option.

**Pull request merged**